### PR TITLE
Psalm integration

### LINKED ISSUE #32: Psalm integration
### Feature Request

|    Q        |   A
|------------ | ------
| QA    | yes

#### Summary

As decided during the [Technical-Steering-Committee Meeting on August 3rd, 2020](https://github.com/laminas/technical-steering-committee/blob/main/meetings/minutes/2020-08-03-TSC-Minutes.md#vote-on-static-analysis-tooling), Laminas wants to implement [vimeo/psalm](https://github.com/vimeo/psalm) in all packages.

Implementing psalm is quite easy.

#### Required
- [ ] Create a `.psalm.xml.dist` in the project root
- [ ] Copy and paste the contents from this [psalm.xml.dist](https://github.com/mezzio/repo-template/blob/6124090610ef1c6d6b0c80a063cafacc3ec68f38/psalm.xml.dist)
- [ ] Run `$ composer require vimeo/psalm`
- [ ] Run `$ vendor/bin/psalm --set-baseline=psalm-baseline.xml`
- [ ] Add a composer script `static-analysis` with the command `psalm --shepherd --stats`
- [ ] Add a new line to `script:` in `.travis.yml`: `- if [[ $TEST_COVERAGE == 'true' ]]; then composer static-analysis ; fi`
- [ ] Remove phpstan from the project (`phpstan.neon.dist`, `.travis.yml` entry, `composer.json` `require-dev` and `scripts`)

##### Optional
- [ ] Fix as many psalm errors as possible.

### PR DESCRIPTION
**Pull request by simonefuoco**
Fixes #32 - psalm installed and configured

|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | yes

### Description

Fixes Issue #32

### CODE REVIEW

**Comment by simonefuoco** [comment]
**Required**

- [x] Create a `psalm.xml.dist` in the project root
- [x] Copy and paste the contents from this [`psalm.xml.dist`](https://github.com/mezzio/repo-template/blob/6124090610ef1c6d6b0c80a063cafacc3ec68f38/psalm.xml.dist)
- [x] Run `$composer require vimeo/psalm`
- [x] Run `$vendor/bin/psalm --set-baseline=psalm-baseline.xml`

> **Notes:**
> - _commented the following line_ `<directory name="bin"/>` in `psalm.xml.dist` _because if **bin** folder is not present, **psalm** command throws an error_
> - _installed  `psalm/plugin-phpunit ` because dependency is listed in `psalm.xml.dist` but not installed so **psalm** command throws an error_

- [x] Add a composer script `static-analysis` with the command `psalm --shepherd --stats`
- [x] Add a new line to script: in `travis.yml`: `if [[ $TEST_COVERAGE == 'true' ]]; then composer static-analysis ; fi`

> **Note:** _`travis.yml` not exists._

- [x]  Remove **phpstan** from the project (`phpstan.neon.dist`, `travis.yml` entry, `composer.json` require-dev and scripts)
> **Notes:**
> - _`phpstan.neon.dist` not exists._
> - _`travis.yml` not exists._
> - _`composer.json` require-dev and scripts do not mention **phpstan**._

**Optional**
- [ ] Fix as many **psalm** errors as possible.
> **Note:** _Doing._

File: psalm.xml.dist [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0"?>
+<psalm
+    totallyTyped="true"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xmlns="https://getpsalm.org/schema/config"
+    xsi:schemaLocation="https://getpsalm.org/schema/config vendor/vimeo/psalm/config.xsd"
+    errorBaseline="psalm-baseline.xml"
+>
+    <projectFiles>
+        <!--<directory name="bin"/>-->
```
Comment (Ocramius) [inline psalm.xml.dist:10]: To be dropped?

File: composer.lock [hunk 2]
```diff
@@ -134,72 +134,132 @@
     ],
     "packages-dev": [
         {
-            "name": "container-interop/container-interop",
-            "version": "1.2.0",
+            "name": "amphp/amp",
+            "version": "v2.6.2",
```
Comment (Ocramius) [inline composer.lock:138]: I think we need #51 before merging this: that would simplify `composer.lock` changes, and prevent the failures that you observed here (incompatible dependencies)

**Review by Ocramius — CHANGES_REQUESTED** [review]

**Comment by Ocramius** [comment]
A rebase is needed, then we should be able to release `2.13.0` :)

**Comment by Ocramius** [comment]
@simonefuoco are you still working on this, or should somebody else rebase?

**Comment by gsteel** [comment]
Thanks for the contribution @simonefuoco - this has been rebased in another patch 👍

**Pull request closed without merging**